components-core/src/main/java/com/adyen/checkout/components/core/internal/analytics/DefaultAnalyticsManager.kt

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2024 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 27/2/2024.
+ */
+
+package com.adyen.checkout.components.core.internal.analytics
+
+import com.adyen.checkout.components.core.internal.analytics.data.NewAnalyticsRepository
+import com.adyen.checkout.components.core.internal.ui.model.AnalyticsParams
+import com.adyen.checkout.components.core.internal.ui.model.AnalyticsParamsLevel
+import com.adyen.checkout.core.AdyenLogLevel
+import com.adyen.checkout.core.internal.util.adyenLog
+import com.adyen.checkout.core.internal.util.runSuspendCatching
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+import kotlin.time.Duration.Companion.seconds
+
+internal class DefaultAnalyticsManager internal constructor(
+    private val analyticsRepository: NewAnalyticsRepository,
+    private val analyticsParams: AnalyticsParams,
+    private val coroutineDispatcher: CoroutineDispatcher = Dispatchers.IO,
+) : AnalyticsManager {
+
+    private var checkoutAttemptId: String? = null
+
+    private var isInitialized: Boolean = false
+
+    private var _coroutineScope: CoroutineScope? = null
+    private val coroutineScope: CoroutineScope get() = requireNotNull(_coroutineScope)
+
+    private var timerJob: Job? = null
+
+    private var ownerReference: String? = null
+
+    override fun initialize(owner: Any, coroutineScope: CoroutineScope) {
+        if (isInitialized) return
```

Comment:
for cases like this where we do nothing logging what happened is a good idea to debug issues in the future